src/ios/UAirshipPlugin.m

```
@@ -819,7 +819,18 @@ - (void)editChannelAttributes:(CDVInvokedUrlCommand *)command {
 
             if ([action isEqualToString:@"set"]) {
                 id value = operation[@"value"];
-                if ([value isKindOfClass:[NSString class]]) {
+                NSString *valueType = operation[@"type"];
+                if ([valueType isEqualToString:@"date"]) {
```

why not use the type everywhere?

---

The number and string types pass through the bridge just fine. It seemed unnecessary.

---

Its consistent in behavior and we could make assumptions about type instead of checking if its an instance of a type.

---

FYI: I left the instance type checks in there as a default if the value type is missing or unknown. This will help if we missed any JS types that used to work "automagically".

---

I would rather throw an exception in JS or toString the value and set it as the atttribute